diff --git a/Programs/Preprocessors/autosp/source/autosp.c b/Programs/Preprocessors/autosp/source/autosp.c
index 6a6b3fa..666bf42 100644
--- a/Programs/Preprocessors/autosp/source/autosp.c
+++ b/Programs/Preprocessors/autosp/source/autosp.c
@@ -2007,7 +2007,7 @@ void process_command (char **ln)
          || prefix ("\\end%", *ln) 
          || prefix ("\\end{document}", *ln) )
   {
-    fprintf (outfile, *ln);
+    fprintf (outfile, "%s", *ln);
     exit(0);
   }
 
