diff --git a/Admin/Defaults/CMakeLists.txt b/Admin/Defaults/CMakeLists.txt
index 8b36fc4..ce03e61 100644
--- a/Admin/Defaults/CMakeLists.txt
+++ b/Admin/Defaults/CMakeLists.txt
@@ -44,8 +44,8 @@ foreach(d ${all_defaults_cmake})
   add_custom_command(
     OUTPUT
       ${CMAKE_BINARY_DIR}/include/miktex/${fn}.defaults.h
-    COMMAND
-      cfg
+    COMMAND ${CMAKE_COMMAND} -E env LD_LIBRARY_PATH=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+      ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/cfg
       --print-classes
       ${CMAKE_CURRENT_BINARY_DIR}/${fn}.defaults
       > ${fn}.defaults.h
