diff --git a/BuildUtilities/cweb/initcommon.c b/BuildUtilities/cweb/initcommon.c
index 3ae9383..99d4189 100644
--- a/BuildUtilities/cweb/initcommon.c
+++ b/BuildUtilities/cweb/initcommon.c
@@ -1107,7 +1107,7 @@ void
  fatal(const char*s,const char*t)
 
 {
-if(*s)printf(s);
+if(*s)printf("%s", s);  // This file *has* to be auto-generated from something...
 err_print(t);
 history= fatal_message;exit(wrap_up());
 }
diff --git a/BuildUtilities/cweb/source/common.w b/BuildUtilities/cweb/source/common.w
index b9d95d5..fe7a81f 100644
--- a/BuildUtilities/cweb/source/common.w
+++ b/BuildUtilities/cweb/source/common.w
@@ -1180,7 +1180,7 @@ concatenated to print the final error message.
 fatal(s,t)
   char *s,*t;
 {
-  if (*s) printf(s);
+  if (*s) printf("%s", s);
   err_print(t);
   history=fatal_message; exit(wrap_up());
 }
diff --git a/BuildUtilities/cweb/source/cweave.w b/BuildUtilities/cweb/source/cweave.w
index 1e73a7c..aaab9f7 100644
--- a/BuildUtilities/cweb/source/cweave.w
+++ b/BuildUtilities/cweb/source/cweave.w
@@ -1784,7 +1784,7 @@ void
 print_cat(c) /* symbolic printout of a category */
 eight_bits c;
 {
-  printf(cat_name[c]);
+  printf("%s", cat_name[c]);
 }
 
 @ The token lists for translated \TEX/ output contain some special control
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 766625b..946aeb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2077,6 +2077,7 @@ if(WITH_BOOTSTRAPPING)
     set(remote_repository
       "${MIKTEX_DEFAULT_REPOSITORY}next/"
     )
+    message( FATAL_ERROR "MIKTEX_RELEASE_STATE=1 not supported with this patch for Nixpkgs!" )
   else()
     set(remote_repository
       "${MIKTEX_DEFAULT_REPOSITORY}"
@@ -2098,7 +2099,7 @@ if(WITH_BOOTSTRAPPING)
       OUTPUT
         ${CMAKE_CURRENT_BINARY_DIR}/${f}
       COMMAND
-        ${CURL_EXECUTABLE} -o ${CMAKE_CURRENT_BINARY_DIR}/${f} -fsSL ${remote_repository}/${f}
+        cp ${CMAKE_SOURCE_DIR}/../${f} ${CMAKE_CURRENT_BINARY_DIR}/${f}
     )
     list(APPEND generated_bootstrapping_files ${CMAKE_CURRENT_BINARY_DIR}/${f})
   endforeach()
diff --git a/Documentation/CMakeLists.txt b/Documentation/CMakeLists.txt
index 20b29fa..50274fe 100644
--- a/Documentation/CMakeLists.txt
+++ b/Documentation/CMakeLists.txt
@@ -26,7 +26,7 @@ set(PROMPTUNX "&#36; ")
 set(PROMPTWIN "C:\&gt; ")
 
 if(MIKTEX_UNIX_ALIKE)
-  set(COMMONINSTALL "/usr/local/share/miktex-texmf")
+  set(COMMONINSTALL "${CMAKE_INSTALL_PREFIX}/share/miktex-texmf")
   set(USERDATA "~/.miktex/texmfs/data")
   set(USERINSTALL "~/.miktex/texmfs/install")
   set(EXESUFFIX "")
diff --git a/Programs/Preprocessors/autosp/source/autosp.c b/Programs/Preprocessors/autosp/source/autosp.c
index 6a6b3fa..666bf42 100644
--- a/Programs/Preprocessors/autosp/source/autosp.c
+++ b/Programs/Preprocessors/autosp/source/autosp.c
@@ -2007,7 +2007,7 @@ void process_command (char **ln)
          || prefix ("\\end%", *ln) 
          || prefix ("\\end{document}", *ln) )
   {
-    fprintf (outfile, *ln);
+    fprintf (outfile, "%s", *ln);
     exit(0);
   }
 
diff --git a/Programs/TeXAndFriends/omega/otps/source/outocp.c b/Programs/TeXAndFriends/omega/otps/source/outocp.c
index fd3fe2c..bbc31f9 100644
--- a/Programs/TeXAndFriends/omega/otps/source/outocp.c
+++ b/Programs/TeXAndFriends/omega/otps/source/outocp.c
@@ -90,7 +90,7 @@ char *otp_names[] = {
 void
 ctp_abort P1C(string, s)
 {
-  fprintf(stderr, s);
+  fprintf(stderr, "%s", s);
   exit(EXIT_FAILURE);
 }
  
diff --git a/cmake/modules/RuntimePaths.cmake b/cmake/modules/RuntimePaths.cmake
index 5b7431f..984cd85 100644
--- a/cmake/modules/RuntimePaths.cmake
+++ b/cmake/modules/RuntimePaths.cmake
@@ -29,7 +29,7 @@ endif()
 
 if(MIKTEX_UNIX_ALIKE)
   set(MIKTEX_SYSTEM_ETC_FONTS_CONFD_DIR "/etc/fonts/conf.d" CACHE PATH "System-wide directory for font configuration data.")
-  set(MIKTEX_SYSTEM_LINK_TARGET_DIR "/usr/local/bin" CACHE PATH "System-wide directory in which to create symbolic links to MiKTeX binaries.")
+  set(MIKTEX_SYSTEM_LINK_TARGET_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "System-wide directory in which to create symbolic links to MiKTeX binaries.")
   set(MIKTEX_SYSTEM_VAR_CACHE_DIR "/var/cache" CACHE PATH "Directory for application cache data.")
   set(MIKTEX_SYSTEM_VAR_LIB_DIR "/var/lib" CACHE PATH "Directory for state information.")
   set(MIKTEX_SYSTEM_VAR_LOG_DIR "/var/log" CACHE PATH "Directory for log files.")
