diff --git a/CMakeLists.txt b/CMakeLists.txt
index 766625b..946aeb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2077,6 +2077,7 @@ if(WITH_BOOTSTRAPPING)
     set(remote_repository
       "${MIKTEX_DEFAULT_REPOSITORY}next/"
     )
+    message( FATAL_ERROR "MIKTEX_RELEASE_STATE=1 not supported with this patch for Nixpkgs!" )
   else()
     set(remote_repository
       "${MIKTEX_DEFAULT_REPOSITORY}"
@@ -2098,7 +2099,7 @@ if(WITH_BOOTSTRAPPING)
       OUTPUT
         ${CMAKE_CURRENT_BINARY_DIR}/${f}
       COMMAND
-        ${CURL_EXECUTABLE} -o ${CMAKE_CURRENT_BINARY_DIR}/${f} -fsSL ${remote_repository}/${f}
+        cp ${CMAKE_SOURCE_DIR}/../${f} ${CMAKE_CURRENT_BINARY_DIR}/${f}
     )
     list(APPEND generated_bootstrapping_files ${CMAKE_CURRENT_BINARY_DIR}/${f})
   endforeach()
