diff --git a/Documentation/CMakeLists.txt b/Documentation/CMakeLists.txt
index 20b29fa..c08f6d5 100644
--- a/Documentation/CMakeLists.txt
+++ b/Documentation/CMakeLists.txt
@@ -196,8 +196,8 @@ set(manual_listings
 add_custom_command(
   OUTPUT
     ${CMAKE_CURRENT_BINARY_DIR}/filetypes
-  COMMAND
-    findtexmf --list-file-types > ${CMAKE_CURRENT_BINARY_DIR}/filetypes
+  COMMAND ${CMAKE_COMMAND} -E env LD_LIBRARY_PATH=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/findtexmf --list-file-types > ${CMAKE_CURRENT_BINARY_DIR}/filetypes
   DEPENDS
     findtexmf
   VERBATIM
